python_modules/dagster/dagster/_utils/source_position.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Mapping, NamedTuple, Optional, Sequence, Union, cast
+
+KeyPathSegment = Union[str, int]
+KeyPath = Sequence[KeyPathSegment]
+
+
+class LineCol(NamedTuple):
+    line: int
+    col: int
+
+
+class SourcePosition(NamedTuple):
+    filename: str
+    start: LineCol
+    end: LineCol
+
+    def __str__(self):
+        return f"{self.filename}:{self.start.line}"
+
+
+class SourcePositionTreeNode(NamedTuple):
+    position: SourcePosition
```

Comment:
nit: docstring would be nice, the meaning/purpose of this class is not super clear from its name

Reply:
in particular what is the "descendent" or "child" of a source position?

Reply:
👍 added a short docstring here and more contextual docstring elsewhere too.